examples/ffi/array_queue.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
```

Comment:
I've only tested running this with `valgrind`(`leaks` on Mac). Any other suggestions for testing?

Reply:
fuzzing?

Reply:
Added fuzz testing on rust side.